### PR TITLE
Document how to interrupt `PacketPeerUDP.wait()`

### LINKED ISSUE #36335: No way to interrupt PacketPeerUDP.wait()
**Godot version:** v3.2.stable.official

**OS/device including version:** macOS 10.15.3

**Issue description:**
I'm starting a new thread with `PacketPeerUDP` listening to specific port with `wait`. Unfortunately there is no way to tell it to stop because I want it to listen only in the right
 time. I tried
- Calling `close`
- Calling `listen` with other port

**Steps to reproduce:**
```gdscript
var thread = Thread.new()
var socket = PacketPeerUDP.new()

func _ready():
    thread.start(self, "_do_on_bg")
    yield(get_tree().create_timer(1), "timeout") # Let the thread settle
    socket.close()
    socket.listen(0)
    thread.wait_to_finish() # Deadlocked forever

func _do_on_bg(_null):
    if socket.listen(789) != OK:
        print("Failed")
        return
    
    while socket.wait() == OK:
        print("Received connection")
```

**Minimal reproduction project:**
[wait-repro.zip](https://github.com/godotengine/godot/files/4222143/wait-repro.zip)

### PR DESCRIPTION
**Pull request by Calinou**
This closes #36335.

### CODE REVIEW

File: doc/classes/PacketPeerUDP.xml [hunk 1]
```diff
@@ -123,6 +123,18 @@
...
+				# Server
+				socket.set_dest_address("127.0.0.1", 789)
+				socket.put_packet("Time to stop".to_ascii())
+
+				# Client
+				while socket.wait() == OK:
+				    var data = socket.get_packet().get_string_from_ascii()
+				    if data == "Time to stop":
+				        return
+				[/codeblock]
```
Comment (Calinou) [inline doc/classes/PacketPeerUDP.xml:137]: @jonbonazza Is this correct and easy enough to understand?

Comment (jonbonazza) [reply doc/classes/PacketPeerUDP.xml:137]: Looks okay to me

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.4.

**Pull request merged**